packages/marko-web-inquiry/routers/submit.js

```diff
@@ -51,6 +51,15 @@ module.exports = ({ queryFragment, notification, confirmation }) => asyncRoute(a
     return true;
   };
 
+  const validatePayload = async () => {
+    if (!payload || !payload.email === '') {
+      error('Form validation failed!', payload);
+      throw exception('Invalid form submission');
+    }
+    return true;
+  };
+
+  await validatePayload();
```

Comment:
There's no need for this to be defined as a separate function or be async, just run the conditional inline. Also please use `!==` instead of `!var === val` :)

Reply:
Moved inline, and I didn't mean to include that ! so I removed it and left the ===

I want it to reject the form if there is no payload, or email is empty in it - from the looks of the code, payload will always exist, even if a bunch of keys with empty string values, but I figured since I don't know how the submission is being triggered, might as well check for payload before I try to reference elements off it just in case.

Reply:
Sounds good! Normally that would be a good use case for the `getAsObject` utility from base-cms/object-path. However, since the `payload` is being destructured from the body object, it's guaranteed to be an object.